MfciPkg/UnitTests/MfciPolicyParsingUnitTest/data/certs/CreateCertificates.ps1

```diff
@@ -0,0 +1,84 @@
+Export-Certificate -Cert $CA -FilePath $currentdir\CA.cer -Type cer
+
+#
+# Create a leaf test key, which will be used for signing Mfci firmware updates
+# certificate is 3072 bytes (maximum available for singing mfci packets)
+# certificate is good for 30 months
+# certificat is exportable, uses sha256 hash function
+# cetificate is signed by CA
+# Text Extensions - EKU Any purpose (2.5.29.37)
+# include EKU  1.3.6.1.5.5.7.3.3, which signifies that certificate can be used for code signing 
+# include EKU 1.3.6.1.4.1.311.45.255.255 which is the EKU that is used by Mfci as an additional check when verifying certificate used to sign packet (this needs to match PcdMfciPkcs7RequiredLeafEKU) 
+$LeafTest = New-SelfSignedCertificate -Subject "FwPolicy test leaf" -KeyLength  3072 -NotAfter (Get-Date).AddMonths(30) -HashAlgorithm sha256 -Signer $CA -KeyExportPolicy Exportable -KeySpec Signature -CertStoreLocation "cert:\LocalMachine\My" -KeyUsageProperty Sign -TextExtension @("2.5.29.37={text}1.3.6.1.4.1.311.45.255.255,1.3.6.1.5.5.7.3.3") -KeyUsage DigitalSignature,KeyEncipherment
+
+# Export the CA keys to a Pfx (Personal Information Exchange)
+Export-PfxCertificate -Cert $LeafTest -ChainOption BuildChain -FilePath $currentdir\Leaf-test.pfx -Password $mypwd
+
+# Export the Public key portion of the certificate
+Export-Certificate -Cert $LeafTest -FilePath $currentdir\Leaf-test.cer -Type cer
+
+
+## Other Test Certificates outside of the main chain.
+
+#
+# Create a different CA, still signed by Root CA, but f
+#
+$CaNotTrusted = New-SelfSignedCertificate -Subject "Not Trusted UEFI Test CA"  -KeyLength  4096 -NotAfter (Get-Date).AddMonths(150) -HashAlgorithm sha256 -Signer $RootCert -KeyExportPolicy Exportable -CertStoreLocation "cert:\LocalMachine\My" -KeyUsageProperty Sign
+
+# Export the CA keys to a Pfx (Personal Information Exchange)
+Export-PfxCertificate -Cert $CaNotTrusted -FilePath $currentdir\CA-NotTrusted.pfx -Password $mypwd
+
+# Export the Public key portion of the certificate
+Export-Certificate -Cert $CaNotTrusted -FilePath $currentdir\CA-NotTrusted.cer -Type cer
+
+
+#
+# Create a leaf test key, with an unmatched EKU
+# 
+$LeafTestBadEku = New-SelfSignedCertificate -Subject "Another test leaf With unmatched EKU" -KeyLength  3072 -NotAfter (Get-Date).AddMonths(30) -HashAlgorithm sha256 -Signer $CA -KeyExportPolicy Exportable -KeySpec KeyExchange -CertStoreLocation "cert:\LocalMachine\My" -KeyUsageProperty Sign -TextExtension @("2.5.29.37={text}1.3.6.1.4.1.311.255.0.0,1.3.6.1.5.5.7.3.3") 
+
+# Export the CA keys to a Pfx (Personal Information Exchange)
+Export-PfxCertificate -Cert $LeafTestBadEku -FilePath $currentdir\Leaf-NoEku.pfx -Password $mypwd
+
+# Export the Public key portion of the certificate
+Export-Certificate -Cert $LeafTestBadEku -FilePath $currentdir\Leaf-NoEku.cer -Type cer
+
+
+#Remove-Item $RootCert
+#Remove-Item $CA
+#Remove-Item $LeafTest
+#Remove-Item $CaNotTrusted
+#Remove-Item $LeafTestBadEku
```

Comment:
Are these lines intended to be checked in commented out?

Reply:
Yes. Power shell installs the certificates into the local system.  Running the power shell script repeatedly will result in multiple versions of the certificates being in the local system's certificate store.  The commented-out lines are a reminder that the local system has the certificates still in the local system store, and that the lines can be used to remote the certs after the script is run.

Reply:
do you think it might make sense to write a comment out to the command shell to tell user how that information and provde a way to remove them.  You could probably wrap that in a conditional and ask user if they want to remove them.

Reply:
Yes, that will be a better method to make it clearer that there are artifacts still in their system.